flutter.el

```diff
@@ -261,6 +261,38 @@ args."
    args
    (display-buffer buffer)))
 
+;; XXX: Copied from s.el
+(defun flutter-s-replace (old new s)
+  "Replaces OLD with NEW in S."
+  (declare (pure t) (side-effect-free t))
+  (replace-regexp-in-string (regexp-quote old) new s t t))
+
+(defun flutter-devices ()
+  "Return a list of devices in strings."
+  (let ((output (shell-command-to-string "flutter devices"))
+        (matches))
+    (with-temp-buffer
+      (insert output)
+      (goto-char (point-min))
+      (while (search-forward " • " nil t)
+        (let ((start (point))
+              (match))
+          (when (search-forward " •" nil t)
+            (setq match (buffer-substring start (point))
+                  match (flutter-s-replace "•" "" match)
+                  match (string-trim match))
+            (push match matches)))
+        (forward-line 1)
+        (beginning-of-line)))
```

Comment:
Rather than parse the human-readable output (which is somewhat likely to change in the future) I would suggest using `flutter devices --machine`.

Reply:
Didn't know that exists! Done!